src/tests/util.rs

```diff
@@ -511,3 +511,106 @@ pub fn generate_key_pair(
     }
     Ok((pub_key, pri_key))
 }
+
+struct CheckTemplate<'a> {
+    bytes: Vec<(CK_ATTRIBUTE_TYPE, Vec<u8>, &'a [u8])>,
+    longs: Vec<(CK_ATTRIBUTE_TYPE, CK_ULONG, CK_ULONG)>,
+    bools: Vec<(CK_ATTRIBUTE_TYPE, CK_BBOOL, CK_BBOOL)>,
+    tmpl: Vec<CK_ATTRIBUTE>,
+}
+
+pub fn check_attributes(
+    session: CK_SESSION_HANDLE,
+    handle: CK_OBJECT_HANDLE,
+    ulongs: &[(CK_ATTRIBUTE_TYPE, CK_ULONG)],
+    bools: &[(CK_ATTRIBUTE_TYPE, bool)],
+    bytes: &[(CK_ATTRIBUTE_TYPE, &[u8])],
+) -> Option<String> {
+    let capacity = ulongs.len() + bytes.len() + bools.len();
+    let mut template = CheckTemplate {
+        bytes: Vec::with_capacity(bytes.len()),
+        longs: Vec::with_capacity(ulongs.len()),
+        bools: Vec::with_capacity(bools.len()),
+        tmpl: Vec::<CK_ATTRIBUTE>::with_capacity(capacity),
+    };
+    for u in ulongs {
+        template.longs.push((u.0, 0, u.1));
+        if let Some(t) = template.longs.last() {
+            template.tmpl.push(make_attribute!(
+                t.0,
+                std::ptr::addr_of!(t.1),
+                CK_ULONG_SIZE
+            ));
+        }
+    }
+    for b in bytes {
+        template.bytes.push((b.0, vec![0; b.1.len()], b.1));
+        if let Some(t) = template.bytes.last() {
+            template.tmpl.push(make_attribute!(
+                t.0,
+                t.1.as_ptr() as *const _,
+                t.1.len()
+            ));
+        }
+    }
+    for b in bools {
+        template
+            .bools
+            .push((b.0, 0, if b.1 { CK_TRUE } else { CK_FALSE }));
+        if let Some(t) = template.bools.last() {
+            template.tmpl.push(make_attribute!(
+                t.0,
+                std::ptr::addr_of!(t.1),
+                std::mem::size_of::<CK_BBOOL>()
```

Comment:
we have this size shortcut defined already so lets use it.
```suggestion
                CK_BBOOL_SIZE
```

Reply:
ok